x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/duration.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { termQuery } from '@kbn/observability-plugin/server';
+import {
+  FAAS_BILLED_DURATION,
+  FAAS_ID,
+  TRANSACTION_DURATION_HISTOGRAM,
+} from '../../../../../common/elasticsearch_fieldnames';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import {
+  fetchAndTransformMetrics,
+  GenericMetricsChart,
+} from '../../fetch_and_transform_metrics';
+import { ChartBase } from '../../types';
+
+const billedDurationAvg = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.billedDurationAvg', {
+    defaultMessage: 'Billed Duration',
+  }),
+};
+
+const transactionDurationAvg = {
+  title: i18n.translate(
+    'xpack.apm.agentMetrics.serverless.transactionDuration',
+    { defaultMessage: 'Transaction Duration' }
+  ),
+};
+
+const chartBase: ChartBase = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.avgDuration', {
+    defaultMessage: 'Avg. Duration',
+  }),
+  key: 'avg_duration',
+  type: 'linemark',
+  yUnit: 'time',
+  series: {},
+};
+
+export async function getDuration({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  faasId,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  faasId?: string;
+  start: number;
+  end: number;
+}): Promise<GenericMetricsChart> {
+  const options = {
+    environment,
+    kuery,
+    setup,
+    serviceName,
+    start,
+    end,
+  };
+
+  const [billedDurationMetrics, transactionDurationMetrics] = await Promise.all(
+    [
+      fetchAndTransformMetrics({
+        ...options,
+        chartBase: { ...chartBase, series: { billedDurationAvg } },
+        aggs: {
+          billedDurationAvg: { avg: { field: FAAS_BILLED_DURATION } },
+        },
+        additionalFilters: [
+          { exists: { field: FAAS_BILLED_DURATION } },
+          ...termQuery(FAAS_ID, faasId),
+        ],
+        operationName: 'get_billed_duration',
+      }),
+      fetchAndTransformMetrics({
+        ...options,
+        chartBase: { ...chartBase, series: { transactionDurationAvg } },
+        aggs: {
+          transactionDurationAvg: {
+            avg: { field: TRANSACTION_DURATION_HISTOGRAM },
```

Comment:
same as the earlier comment - should we use `getDurationFieldForTransactions` here?